master/internal/core.go

```diff
@@ -267,7 +274,18 @@ func (m *Master) rwCoordinatorWebSocket(socket *websocket.Conn, c echo.Context)
 	return actorRef.AwaitTermination()
 }
 
-func (m *Master) initializeResourceProviders(proxyRef *actor.Ref, provisionerSlotsPerInstance int) {
+func (m *Master) initializeResourceProviders(proxyRef *actor.Ref, cert *tls.Certificate, provisionerSlotsPerInstance int) {
+	var certBytes []byte
+	if cert != nil {
+		for _, c := range cert.Certificate {
+			b := pem.EncodeToMemory(&pem.Block{
+				Type:  "CERTIFICATE",
+				Bytes: c,
+			})
+			certBytes = append(certBytes, b...)
```

Comment:
I think if we move the `pem.EncodeToMemory` logic to `masterCertArchive` in `master/pkg/tasks/copy.go` and pass a `*tls.Certificate` named `MasterCert` all the way down as opposed to `[]bytes`/`certBytes`, it feels a bit less awkward.

Mainly two reasons I say this: one it seems like the enconding belongs in the `masterCertArchive` more than `initializeResourceProviders` and passing a `[]byte` around so far feels like it needs better typing (though the name `certBytes` helps, but cert for what?).

These are just my opinions, since you specifically requested feedback about this.